src/main/java/org/jacop/constraints/XneqY.java

```diff
@@ -78,6 +78,11 @@ public XneqY(IntVar x, IntVar y) {
     }
 
     @Override public void consistency(final Store store) {
+        if (x == y) {
```

Comment:
The purpose of consistency function as well as notConsistency function is to perform pruning of domains. 
In this case, however a much better approach would be to throw FailException to indicate immediately that there is no solution. We do not do call domain.clear() to indicate to solver that there is no solution.

Reply:
throw store.failException; 
instead of 
x.domain.clear();
return;

Reply:
Ah, so the exception is intended to be used for that purpose. Changed!